scripts/quests/outlands/Like_Shining_Leggings.lua

```diff
@@ -0,0 +1,105 @@
+local tradeTrackingVar = "Prog";
+local requiredLeggings = 10
+local successfulTradeNum = 0
+
+quest.reward =
+{
+    fameArea = xi.quest.fame_area.NORG,
+    fame     = 100,
+    title    = xi.title.LOOKS_GOOD_IN_LEGGINGS,
+    item     = xi.items.SCROLL_OF_DOKUMORI_ICHI,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return (status == QUEST_AVAILABLE and player:getFameLevel(xi.quest.fame_area.NORG) >= 3)
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Heizo'] = quest:progressEvent(127),
+
+            onEventFinish =
+            {
+                [127] = function(player, csid, option, npc)
+                    quest:begin(player) -- Start Like Shining Leggings
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Heizo'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(128, quest:getVar(player, tradeTrackingVar))-- Update the player on the number of Rusty Leggings they have turned in
+                end,
+
+                onTrade = function(player, npc, trade)
+                    local curNumTradedIn = quest:getVar(player, tradeTrackingVar)
+                    local numTradingInNow = trade:getItemQty(xi.items.RUSTY_LEGGINGS);
+                    local remainderNeeded = utils.clamp(requiredLeggings - curNumTradedIn, 0, requiredLeggings);
+
+                    local clampedTrade = math.min(numTradingInNow, remainderNeeded) -- Only take the amount needed to complete the quest and nothing more
+
+                    if npcUtil.tradeHas(trade, { { xi.items.RUSTY_LEGGINGS, clampedTrade } }) then
+                        local totalWithTrade = curNumTradedIn + clampedTrade
+                        successfulTradeNum = totalWithTrade
+
+                        if totalWithTrade >= 10 then
+                            return quest:progressEvent(129) -- Complete the quest
+                        elseif totalWithTrade <= 9 then
+                            return quest:event(128, totalWithTrade)
+                        end
+                    else
+                        successfulTradeNum = 0
+                        return quest:event(128, curNumTradedIn)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [128] = function(player, csid, option, npc)
+                    if successfulTradeNum > 0 then
+                        quest:setVar(player, tradeTrackingVar, successfulTradeNum)-- Keep track of traded so far
+                        player:confirmTrade()
+                    else
+                        player:tradeComplete(false) -- There were extra items in the trade
+                    end
+                end,
```

Comment:
Is there dialog that is shown if more are traded? Or does retail just not accept it?

Reply:
The comment actually could be updated. It was leftover from when I was using "tradeHasOnly" which would fail if you included anything other than the subligar or leggings. 
I switched to "tradeHas" which only takes the desired items and also will not take more than is required.
I'm pretty sure the trade will only fail now if you don't include any of the desired item.

Reply:
I meant if you traded 8 on first go and then 3 on second trade, was there a dialog that said something like, "I only require 10" and then returned the 3.. but the dumps doesn't appear to have that dialog - maybe as an friendly reminder, if it fails, have the `return quest:event(128, curNumTradedIn)` ?

Reply:
Oh I see what you meant; yea I'm not certain on the retail behavior, but that is what the original version of this quest implementation was doing. It should still be doing that currently on line 79. If the trade fails it will display their current progress message.